packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -747,6 +747,83 @@ export class KubernetesClient {
     }
   }
 
+  async readNamespacedDeployment(name: string, namespace: string): Promise<V1Deployment | undefined> {
+    let telemetryOptions = {};
+    const k8sAppsApi = this.kubeConfig.makeApiClient(AppsV1Api);
+    try {
+      const res = await k8sAppsApi.readNamespacedDeployment(name, namespace);
+      if (res?.body) {
+        return res.body;
+      } else {
+        return undefined;
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesReadNamespacedDeployment', telemetryOptions);
+    }
+  }
+  async readNamespacedIngress(name: string, namespace: string): Promise<V1Ingress | undefined> {
+    let telemetryOptions = {};
+    const k8sNetworkingApi = this.kubeConfig.makeApiClient(NetworkingV1Api);
+    try {
+      const res = await k8sNetworkingApi.readNamespacedIngress(name, namespace);
+      if (res?.body) {
+        return res.body;
+      } else {
+        return undefined;
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesReadNamespacedIngress', telemetryOptions);
+    }
+  }
+
+  async readNamespacedRoute(name: string, namespace: string): Promise<V1Route | undefined> {
+    let telemetryOptions = {};
+    const k8sCustomObjectsApi = this.kubeConfig.makeApiClient(CustomObjectsApi);
+    try {
+      const res = await k8sCustomObjectsApi.getNamespacedCustomObject(
+        'route.openshift.io',
+        'v1',
+        namespace,
+        'routes',
+        name,
+      );
+      if (res?.body) {
+        return res.body as V1Route;
+      } else {
+        return undefined;
+      }
+    } catch (error) {
+      telemetryOptions = { error: error };
+      throw this.wrapK8sClientError(error);
+    } finally {
+      this.telemetry.track('kubernetesReadNamespacedRoute', telemetryOptions);
+    }
+  }
+
+  async readNamespacedService(name: string, namespace: string): Promise<V1Service | undefined> {
+    let telemetryOptions = {};
+    const k8sApi = this.kubeConfig.makeApiClient(CoreV1Api);
+    try {
+      const res = await k8sApi.readNamespacedService(name, namespace);
+      if (res?.body) {
+        return res.body;
+      } else {
+        return undefined;
+      }
```

Comment:
```suggestion
return res.body;
```

Reply:
Had to rebase, so I fixed this at the same time, along with the place I copy/pasted from.